libraries/androidutils/src/main/kotlin/io/element/android/libraries/androidutils/uri/UriExtensions.kt

```diff
@@ -12,3 +12,16 @@ import android.net.Uri
 const val IGNORED_SCHEMA = "ignored"
 
 fun createIgnoredUri(path: String): Uri = Uri.parse("$IGNORED_SCHEMA://$path")
+
+fun Uri.setQueryParameter(key: String, value: String): Uri {
+    val existingParams = queryParameterNames
+    return buildUpon().apply {
+        clearQuery()
+        existingParams.forEach { existingKey ->
+            if (existingKey != key) {
+                appendQueryParameter(existingKey, getQueryParameter(existingKey))
+            }
+        }
+        appendQueryParameter(key, value)
+    }.build()
+}
```

Comment:
Only `appendQueryParameter` exists in `Uri.Builder`, so sadly, I had to create this extension.